src/elasticsearch/elasticsearch.ts

```diff
@@ -28,55 +45,25 @@ export function getElasticsearchNode(instance: Instance): Promise<ElasticsearchN
                 throw `expected information about a single node, but got: ${JSON.stringify(json)}`;
             }
         })
-}
+    }
 
-export function getDocuments(node: ElasticsearchNode): Promise<Documents> {
-    return ssmCommand(`curl localhost:9200/_nodes/${node.ec2Instance.privateIp}/stats`, node.ec2Instance.id)
-        .then(response => {
-            const nodeStats: NodeStats = JSON.parse(response);
-            return nodeStats.nodes[node.nodeId].indices.docs;
-        });
-}
+    getDocuments(node: ElasticsearchNode): Promise<Documents> {
+        return this.execute(`/_nodes/${node.ec2Instance.privateIp}/stats`) // node.ec2Instance.id
+            .then((nodeStats: NodeStats) => { return nodeStats.nodes[node.nodeId].indices.docs; });
+    }
 
-export function updateRebalancingStatus(instanceId: string, status: string): Promise<boolean> {
-    const disableRebalancingCommand: object =
-        {
-            "transient": {
-                "cluster.routing.rebalance.enable": status
-            }
-        };
-    const elasticsearchCommand = `curl -X PUT "localhost:9200/_cluster/settings" -H 'Content-Type: application/json' -d '${JSON.stringify(disableRebalancingCommand)}'`;
-    return ssmCommand(elasticsearchCommand, instanceId)
-        .then((rawResponse: string) => {
-            if (!successfulAction(rawResponse)) {
-                throw `unexpected Elasticsearch response when attempting to update re-balancing status, we got: ${rawResponse}`;
-            } else {
-                console.log(`Successfully updated re-balancing status of cluster to: ${status}`);
-                return true;
-            }
-        })
-}
+    setClusterSetting(settingName: string, settingValue: string): Promise<boolean> {
+        return this.executeCommand("/_cluster/settings", {
+            "transient": { settingName: settingValue }
```

Comment:
_"Starting with ECMAScript 2015, the object initializer syntax also supports [computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names). That allows you to put an expression in brackets `[]`, that will be computed and used as the property name."_

```suggestion
            "transient": { [settingName]: settingValue }
```

...otherwise we get JSON generated that has literally `settingName` injected in it as a field, like this:

```json
{
    "transient": {
        "settingName": ""
    }
}
```